client/pages/students.tsx

```diff
@@ -0,0 +1,55 @@
+import { useState, useEffect } from 'react'
+
+// TODO(amantri): get this from the server code, preferably auto-generated
+interface Student {
+  student_id: string;
+  first_name: string;
+  last_name: string;
+  assigned_case_manager: string;
+}
+
+export default function Students() {
+  const [data, setData] = useState(null)
+  const [isLoading, setLoading] = useState(false)
+
+  useEffect(() => {
+    setLoading(true)
+    // TODO(amantri): Get backend url from env
+    fetch('http://localhost:8080/students/list')
```

Comment:
Is fetch the right way to get make the api call, or should I be using axios or something else?

Reply:
there's a ton of data fetching libraries for React
generally I wouldn't use `fetch` directly because you then need to deal with error & loading states yourself (and miss out on a lot of goodies from other libraries)

[React Query](https://tanstack.com/query/latest/docs/react/overview) is a very popular and easy to use data fetching library.

[SWR](https://swr.vercel.app/) is another moderately popular one that doesn't provide quite as much functionality, but maybe a little simpler.

Reply:
Thanks for the pointers. Switched to SWR